integrationTests/multiShard/transaction/sendBulkTransactions_test.go

```diff
@@ -0,0 +1,187 @@
+
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+
+	nodes := integrationTests.CreateNodes(
+		numOfShards,
+		nodesPerShard,
+		numMetachainNodes,
+		integrationTests.GetConnectableAddress(advertiser),
+	)
+
+	nodes[0] = integrationTests.NewTestProcessorNode(uint32(numOfShards), 0, firstSkInShard, integrationTests.GetConnectableAddress(advertiser))
+	integrationTests.CreateAccountForNodes(nodes)
+	integrationTests.DisplayAndStartNodes(nodes)
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Node.Stop()
+		}
+	}()
+
+	numOfCrossShardTxsToSend := 30
+	numOfIntraShardTxsToSend := 20
+	totalNumOfTxs := numOfCrossShardTxsToSend + numOfIntraShardTxsToSend
+	numOfAccounts := 20
+	receivedTxsPerShard = make(map[uint32]int, 0)
+	mutGeneratedTxHashes := sync.Mutex{}
+	//wire a new hook for generated txs on a node in sender shard to populate tx hashes generated
+	for _, n := range nodes {
+		nodeShardId := n.ShardCoordinator.SelfId()
+		if nodeShardId != sharding.MetachainShardId {
+			txPoolRegister(n, &mutGeneratedTxHashes)
+		}
+	}
+
+	// generate a given number of accounts
+	accounts := generateAccounts(numOfAccounts)
+	accountsByShard := make(map[uint32][]keyPair, 0)
+	for _, acc := range accounts {
+		pkBytes, _ := acc.pk.ToByteArray()
+		shardId := nodes[0].ShardCoordinator.ComputeId(state.NewAddress(pkBytes))
+		accountsByShard[shardId] = append(accountsByShard[shardId], acc)
+	}
+
+	var txs []*transaction.Transaction
+
+	// generate cross shard txs from shard 1 to shard 0
+	for len(txs) < numOfCrossShardTxsToSend {
+		// generate a tx from shard 0 to shard 1 and check if signature is correct
+		randomSenderId := getRandomIndex(accountsByShard[1])
+		sender := accountsByShard[1][randomSenderId]
+
+		randomReceiverId := getRandomIndex(accountsByShard[0])
+		receiver := accountsByShard[0][randomReceiverId]
+
+		tx := generateTx(sender.sk, receiver.pk)
+
+		txs = append(txs, tx)
+	}
+
+	// generate intra shard txs in shard 1
+	for len(txs) < totalNumOfTxs {
+		// generate a tx from a random account from shard 1 to other random account from shard 1
+		randomSenderId := getRandomIndex(accountsByShard[1])
+		sender := accountsByShard[1][randomSenderId]
+
+		randomReceiverId := getRandomIndex(accountsByShard[1])
+		receiver := accountsByShard[1][randomReceiverId]
+
+		tx := generateTx(sender.sk, receiver.pk)
+
+		txs = append(txs, tx)
+	}
+
+	sentTxs, err := nodes[0].Node.SendBulkTransactions(txs)
+	assert.Nil(t, err)
+
+	// wait for txs to be sent
+	time.Sleep(1 * time.Second)
+	// check if all txs were sent
+	if len(accountsByShard[0]) > 1 {
+		assert.Equal(t, uint64(totalNumOfTxs), sentTxs)
+	} else {
+		assert.Equal(t, uint64(numOfCrossShardTxsToSend), sentTxs)
+	}
+
+	mutGeneratedTxHashes.Lock()
+	// check if all nodes in shard 0 got cross shard txs in pool
+	assert.Equal(t, numOfCrossShardTxsToSend, receivedTxsPerShard[0]/nodesPerShard)
```

Comment:
It is, because all nodes in a shard will receive those transactions. So a division by the number of nodes per shard is mandatory to see the exact number of txs received by a single node

Reply:
Ok... All nodes register their transactions in the receivedTxsPerShard[shardId] and you have to divide by the number of nodes